simpleclient_spring_web/src/main/java/io/prometheus/client/spring/web/MethodTimer.java

```diff
@@ -0,0 +1,64 @@
+package io.prometheus.client.spring.web;
+
+import io.prometheus.client.Summary;
+import org.apache.commons.lang3.StringUtils;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+
+import java.util.HashMap;
+
+/**
+ * This class automatically times (via aspectj) the execution of methods by their signature, if it's been enabled via {@link EnablePrometheusTiming}
+ * and in methods annotated (or within classes annotated) with {@link PrometheusTimeMethods}
+ *
+ * @author Andrew Stuart <andrew.stuart2@gmail.com>
+ */
+@Aspect
+@ControllerAdvice
+@Component
+public class MethodTimer {
+    public static final String METRIC_NAME = "prometheus_method_timing_seconds";
+
+    public static final HashMap<String, Summary> collectors = new HashMap<String, Summary>(10);
+
+    public static final Summary defaultSummary = Summary.build()
+            .name(METRIC_NAME)
+            .help("Automatic annotation-driven method timing")
+            .labelNames("signature")
+            .register();
+
+    @Around("@annotation(annot)")
+    public Object timeMethod(ProceedingJoinPoint pjp, PrometheusTimeMethods annot) throws Throwable {
+        Summary summary = defaultSummary;
+        String name = annot.value();
+        if (!StringUtils.isEmpty(name)) {
```

Comment:
Alright, I updated this with a ReentrantReadWriteLock since 99% of the time we will probably only be reading the map and this will offer proper multithreaded access for those cases and thus keep this from becoming a bottleneck.

Double (or triple) check my implementation, though, since I'd rather not be introducing deadlocks.